cmd/apiProxyDownload_test.go

```diff
@@ -0,0 +1,73 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type apiProxyDownloadMockUtils struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func TestRunApiProxyDownload(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Successfull Download of API Proxy", func(t *testing.T) {
+		tempDir, tmpErr := ioutil.TempDir("", "")
+		defer os.RemoveAll(tempDir) // clean up
+		if tmpErr != nil {
+			t.Fatal("Failed to create temporary directory")
+		}
+		apiServiceKey := `{
+			"oauth": {
+				"url": "https://demo",
+				"clientid": "demouser",
+				"clientsecret": "******",
+				"tokenurl": "https://demo/oauth/token"
+			}
+		}`
+		config := apiProxyDownloadOptions{
+			APIServiceKey: apiServiceKey,
+			APIProxyName:  "flow1",
+			DownloadPath:  tempDir,
+		}
+		httpClient := httpMockCpis{CPIFunction: "APIProxyDownload", ResponseBody: ``, TestType: "PositiveAndGetetIntegrationArtifactDownloadResBody"}
+		err := runApiProxyDownload(&config, nil, &httpClient)
+		absolutePath := filepath.Join(tempDir, "flow1.zip")
+		assert.Equal(t, fileExists(absolutePath), true)
+		if assert.NoError(t, err) {
+			t.Run("check url", func(t *testing.T) {
```

Comment:
Why are these checks handled differently than the check for the existence of the zip file? I would strongly suggest to pick a style and stick with it, not to mix and match. Unless there is a reason to do so, of course.

Reply:
following same approach as integrationArtifactDownload_test.go command, where code was already reviewed and merged,

Reply:
Please add a link so I don't need to go search